File: C++/Algorithm/Bubble.java
public class Bubble
{
	public static void main(String []args){
		int a[] = {2,5,1,7,8,9};
		//System.out.println(a.length);
		for(int i=0;i<a.length;i++){
			//System.out.print(a[i]+":");
			for(int j=a.length-1;j>0;j--){
				if(a[j]<a[j-1]){
//					a[j] = a[j]^a[j+1];
//					a[j+1] = a[j]^a[j+1];
//					a[j] = a[j]^a[j+1];
					int tmp = a[j];
					a[j] = a[j-1];
					a[j-1] = tmp;
				}
			}
			System.out.println(a[i]);
		}

	}
}

File: C++/Algorithm/bubble.cpp
#include<iostream>
using namespace std;

int main(){
	int a[] ={2,4,5,6,1,7};
	for(int i=0;i<6;i++){
		for(int j=5;j>0;j--){
			if(a[j]<a[j-1]){
				int tmp = a[j];
				a[j] = a[j-1];
				a[j-1] = tmp;
			}
		}
		cout<<a[i]<<endl;
	}
}